lower library verbosity

talg2324 opened the pull request:
1. print(...) -> logging.log(logging.INFO, ...)
2. warnings.warn(...) -> logging.log(logging.WARN, ...)
3. removed linux=True
4. os.system -> subprocess.run
5. Option to disconnect subprocess pipe and hide executable console logging

faridyagubbayli commented on kwave/options/simulation_execution_options.py:
```diff
@@ -34,6 +34,9 @@ class SimulationExecutionOptions:
     verbose_level: int = 0
 
+    # show simulation log
+    show_sim_log: bool = True
```
This variable is not used, what's the reason for introducing it?

talg2324 replied:
used in line 31 of executor.py to pipe the stdout to DEVNULL

faridyagubbayli replied:
Ahh okay thanks!

faridyagubbayli commented on tests/setup_test.py:
```diff
@@ -1,8 +1,9 @@
 PACKAGE_PARENT = '..'
 SCRIPT_DIR = os.path.dirname(os.path.realpath(os.path.join(os.getcwd(), os.path.expanduser(__file__))))
-print(os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
+logging.log(logging.INFO,  os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
```
There are two spaces after `logging.log(logging.INFO,` part in all changes in this PR. Maybe consider using one space instead?

talg2324 replied:
good catch, I'll fix. Thanks

faridyagubbayli reviewed (approved):
LGTM! Left two non-blocking comments. Thanks for working on this!

waltsims reviewed (changes requested):
This looks great. Thank you.

I have left a few non blocking comments. There one breaking changes in the CI, but other than that, this can be merged in.

waltsims commented:
Also if you merge master into your PR, the CI will run for the PR as well.

talg2324 commented on kwave/utils/mapgen.py:
```diff
@@ -337,7 +337,7 @@ def power_law_kramers_kronig(w: np.ndarray, w0: float, c0: float, a0: float, y:
 
     if 0 >= y or y >= 3:
-        warnings.warn("y must be within the interval (0,3)", UserWarning)
+        logging.log(logging.WARN, f"{UserWarning.__name__}: y must be within the interval (0,3)")
```
Sure. It looks a bit awkward.. not sure that pytests can be adapted to work with logging module so maybe this is a long-term advantage to using warnings.warn. 
Should be possible to read the last logging message from the root logger (or module-specific logger when implemented) instead, but that requires sorting the module logger first

talg2324 commented:
Hi, I think I've addressed all your comments.
The only one I didn't implement was Walter's comment on adding the namespace to the logger.
I think it's a good idea, but I'll leave that one to you guys because there's several good ways to implement+configure it and I'm sure you have something in mind.

waltsims reviewed (approved):
Thanks for those changes. Can you please correct the RUFF formatting errors?

once complete we can merge this in:
```
Error: kwave/reconstruction/beamform.py:188:141: E501 Line too long (143 > 140 characters)
Error: tests/diff_utils.py:2:20: F401 `pprint.pprint` imported but unused
```

talg2324 commented:
> Thanks for those changes. Can you please correct the ruff formatting errors?
> 
> once complete we can merge this in:
> 
> ```
> Error: kwave/reconstruction/beamform.py:188:141: E501 Line too long (143 > 140 characters)
> Error: tests/diff_utils.py:2:20: F401 `pprint.pprint` imported but unused
> ```

Can you approve the test workflow?

waltsims commented:
Thanks for the great PR. I've merged this in.

I hope you contribute again.

Best,
Walter
